test/e2e/storage/testsuites/provisioning.go

```diff
@@ -348,21 +379,17 @@ func (t StorageClassTest) TestDynamicProvisioning() *v1.PersistentVolume {
 		}
 	}()
 
-	if class == nil {
-		// StorageClass is nil, so the default one will be used
-		scName, err := e2epv.GetDefaultStorageClassName(client)
-		framework.ExpectNoError(err)
-		defaultSC, err := client.StorageV1().StorageClasses().Get(context.TODO(), scName, metav1.GetOptions{})
+	// ensure that the claim refers to the provisioned StorageClass
+	framework.ExpectEqual(*claim.Spec.StorageClassName, class.Name)
+
+	// if late binding is configured, create and delete a pod to provision the volume
+	if *class.VolumeBindingMode == storagev1.VolumeBindingWaitForFirstConsumer {
```

Comment:
@msau42 the conversation was dropped after I did the rebase fix, sorry about that.

After debugging the test I saw that the Pod created in this if statement was failing with the following error:

```
 Warning  FailedMount  7s    kubelet, e2e-test-mauriciopoppe-minion-group-pnq9  Unable to attach or mount volumes: unmounted volumes=[volume1], unattached volumes=[volume1 default-token-b9jmr[]: volume volume1 has volumeMode Block, but is specified in volumeMounts
```

After some debugging I found out that the test is failing because the claim dynamically provisions a PV with volumeMode = Block and this pod is created with `volumeMounts` instead of `volumeDevices` for all the cases, if the objective of this test is to also make it work with a BlockDevice then I can create either `volumeMounts` (the default) or `volumeDevices` inside `e2e.CreatePod`, we could also skip this test for block devices, what do you think?

Reply:
I opted for adding support to create volumeDevices in the e2e pod creation method